lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -319,7 +319,68 @@ This anti-pattern was formerly known as [Accessing non-existent Map/Struct field
 
 ## Dynamic atom creation
 
-TODO.
+#### Problem
+
+An `atom` is a primary data type of Elixir whose value is its own name. They are often useful to identify resources or express an operation's state. The creation of an `atom` do not characterize an anti-pattern by itself; however, `atoms` are not collected by Elixir's Garbage Collector, so values of this type live in memory while software is executing, during its entire lifetime. Also, BEAM limits the number of `atoms` that can exist in an application (`1_048_576`) and each `atom` has a maximum size limited to 255 Unicode code points. For these reasons, the dynamic atom creation is considered an anti-pattern, since in this way the developer has no control over how many `atoms` will be created during the software execution. This unpredictable scenario can expose the software to unexpected behavior caused by excessive memory usage, or even by reaching the maximum number of `atoms` possible.
+
+#### Example
+
+Imagine that you are implementing a code that performs the conversion of `string` values into `atoms` to identify resources. These `strings` can come from user input or even have been received as response from requests to an API. As this is a dynamic and unpredictable scenario, it is possible for identical `strings` to be converted into new `atoms` that are repeated unnecessarily. This kind of conversion, in addition to wasting memory, can be problematic for the software if it happens too often.
+
+```elixir
+defmodule Identifier do
+  def generate(id) when is_bitstring(id) do
+    String.to_atom(id)  #<= dynamic atom creation!!
+  end
+end
+```
+
+```elixir
+iex> string_from_user_input = "my_id"
+"my_id"
+iex> string_from_API_response = "my_id"
+"my_id"
+iex> Identifier.generate(string_from_user_input)
+:my_id
+iex> Identifier.generate(string_from_API_response)
+:my_id   #<= atom repeated was created!
+```
+
+When we use the `String.to_atom/1` function to dynamically create an `atom`, it is created regardless of whether there is already another one with the same value in memory, so when this happens automatically, we will not have control over meeting the limits established by BEAM.
```

Comment:
I'm going to expose my ignorance here... I don't think this is true? The docs for `String.to_atom/1` and `:erlang.binary_to_atom/1` don't mention anything like this, and it goes against the definition of atoms themselves, so I'd be surprised if this was the case 🤔

Reply:
> I confirm it is not the case. Existing atoms are not created again.

Curious. In my understanding, this is an important point in the "root" of this anti-pattern definition. @josevalim, do you have any suggestions on how to modify this in our text?

Reply:
The issue about String.to_atom is for atoms that do not yet exist. If you are covering an external parameter, such as requests from an API to atom, then someone can use to create enough atoms to cause the system to shut down. Which is why @whatyouhide mentioned APIs being a key issue.